src/Entity/Book.php

```diff
@@ -0,0 +1,163 @@
+<?php
+namespace App\Entity;
+
+use App\Repository\BookRepository;
+use DateTime;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+use App\Validator as CustomAssert;
+
+/**
+ * @ORM\Entity(repositoryClass=BookRepository::class)
+ * @CustomAssert\LeasingConstraint(number=3)
+ */
+class Book
```

Comment:
I discourage the use of setters in general.
If you're creating an Entity, it should be valid and consistent from the moment it is created, otherwise you will get a lot of troubles down the line, if you decide to persist an entity in the inconsistent state.
One way to fix this is to provide all required values in the constructor and define named setters for specific scenarios, like setting an optional dependency.

And the fluent interfaces usually call for very specific use-cases, I'd avoid using them on a general basis.

Reply:
Yeap, it's something to take in consideration. In this casa, maybe a method to the entity like "fromArray" ... to hydrate the entity when is created ?

Reply:
I usually go with a builder method in the service, that manages the entity. I have some input object and a factory method in the service, that consumes the object and spits out the ready entity. But the entity itself receives values exclusively via the constructor.